lean/components/docker/lean_runner.py

```diff
@@ -130,7 +130,20 @@ def run_lean(self,
                 "mode": "rw"
             }
 
-        run_options["commands"].append("exec dotnet QuantConnect.Lean.Launcher.dll")
+        # Setup debugging with QC Extension
+        if debugging_method == DebuggingMethod.QC:
+            run_options["ports"]["5678"] = "0" #Assign a random port
+
+            if lean_config["algorithm-language"] == "CSharp":
+                # CSharp Debugging means starting with netcoredbg in server mode
+                # TODO: Need to delete other pdbs?
```

Comment:
This is requires a change to the Lean image, needs to have netcoredbg installed 👍🏽